includes/civicrm.shortcodes.php

```diff
@@ -295,20 +308,25 @@ public function render_single($atts) {
       }
     }
 
-    // Preprocess shortcode attributes.
-    $args = $this->preprocess_atts($atts);
+    // If there are ACTUAL CiviCRM query vars, let them take priority.
+    if (!$this->civi->civicrm_in_wordpress()) {
 
-    // Sanity check for improperly constructed shortcode.
-    if ($args === FALSE) {
-      return '<p>' . __('Do not know how to handle this shortcode.', 'civicrm') . '</p>';
-    }
+      // Preprocess shortcode attributes.
+      $args = $this->preprocess_atts($atts);
 
-    // invoke() requires environment variables to be set.
-    foreach ($args as $key => $value) {
-      if ($value !== NULL) {
-        set_query_var($key, $value);
-        $_REQUEST[$key] = $_GET[$key] = $value;
+      // Sanity check for improperly constructed shortcode.
+      if ($args === FALSE) {
+        return '<p>' . __('Do not know how to handle this shortcode.', 'civicrm') . '</p>';
```

Comment:
As the area's being touched this message, in the rare circumstances it gets hit isn't actually that useful.

Something like: "The shortcode could not be handled. It could be malformed or being used incorrectly." might be more useful :-)

Reply:
@MikeyMJCO Done.